packages/local_auth/local_auth/android/src/test/java/io/flutter/plugins/localauth/LocalAuthTest.java

```diff
@@ -20,45 +24,95 @@
+  @Mock private DartExecutor mockDartExecutor;
+  @Mock private ActivityPluginBinding mockActivityBinding;
+  @Mock private MethodChannel.Result mockResult;
+  @Mock private FlutterPluginBinding mockPluginBinding;
 
-    Context mockContext = mock(Context.class);
-    when(mockActivity.getBaseContext()).thenReturn(mockContext);
+  @Before
+  public void setUp() {
+    plugin = new LocalAuthPlugin();
+    mockContext = mock(Context.class);
+    mockActivity = mock(Activity.class);
+    mockLifecycle = mock(Lifecycle.class);
+    mockFlutterEngine = mock(FlutterEngine.class);
+    mockDartExecutor = mock(DartExecutor.class);
+    mockActivityBinding = mock(ActivityPluginBinding.class);
+    mockResult = mock(MethodChannel.Result.class);
+    mockPluginBinding = mock(FlutterPluginBinding.class);
 
     final HiddenLifecycleReference mockLifecycleReference = mock(HiddenLifecycleReference.class);
+    when(mockActivity.getApplicationContext()).thenReturn(mockContext);
     when(mockActivityBinding.getLifecycle()).thenReturn(mockLifecycleReference);
-
-    final Lifecycle mockLifecycle = mock(Lifecycle.class);
     when(mockLifecycleReference.getLifecycle()).thenReturn(mockLifecycle);
-
-    final FlutterPluginBinding mockPluginBinding = mock(FlutterPluginBinding.class);
-    final FlutterEngine mockFlutterEngine = mock(FlutterEngine.class);
+    when(mockActivityBinding.getActivity()).thenReturn(mockActivity);
+    when(mockActivity.getBaseContext()).thenReturn(mockContext);
     when(mockPluginBinding.getFlutterEngine()).thenReturn(mockFlutterEngine);
-
-    DartExecutor mockDartExecutor = mock(DartExecutor.class);
     when(mockFlutterEngine.getDartExecutor()).thenReturn(mockDartExecutor);
 
-    final LocalAuthPlugin plugin = new LocalAuthPlugin();
+    plugin.onAttachedToEngine(mockPluginBinding);
+    plugin.onAttachedToActivity(mockActivityBinding);
+  }
+
+  @Test
+  public void isDeviceSupportedReturnsFalse() {
+    plugin.onMethodCall(new MethodCall("isDeviceSupported", null), mockResult);
+    verify(mockResult).success(false);
+  }
+
+  @Test
+  public void onDetachedFromActivity_ShouldReleaseActivity() {
+
     plugin.onAttachedToEngine(mockPluginBinding);
     plugin.onAttachedToActivity(mockActivityBinding);
     assertNotNull(plugin.getActivity());
 
     plugin.onDetachedFromActivity();
     assertNull(plugin.getActivity());
   }
+
+  @Test
+  public void getAvailableBiometrics_ShouldReturnUndefinedBiometrics() {
+    plugin.onMethodCall(new MethodCall("getAvailableBiometrics", null), mockResult);
+    verify(mockResult).success(Collections.singletonList("undefined"));
+  }
+
+  @Test
+  public void getAvailableBiometrics_ShouldReturnEmptyBiometrics() {
+    final BiometricManager mockBiometricManager = mock(BiometricManager.class);
+    when(mockBiometricManager.canAuthenticate()).thenReturn(BiometricManager.BIOMETRIC_SUCCESS);
+    plugin.overrideBiometricManager(mockBiometricManager);
```

Comment:
The preconditions here are unclear as well. Why does `SUCCESS` correspond to nothing? You should be explicitly returning false for the features.

But also, this test is not actually a valid case. The docs say "Returns BIOMETRIC_SUCCESS if a strong biometric can currently be used (enrolled and available)." So you are actually asserting the wrong behavior here, because you've created an invalid scenario, where the mock APIs are giving conflicting answers on enrollment.